src/remote/blobs.go

```diff
@@ -274,40 +274,52 @@ func (c *Client) byteStreamDownloadName(digest *pb.Digest) string {
 // Data is not required.
 // The given function is a callback that receives a channel to send these blobs on; it
 // should close it when finished.
-func (c *Client) downloadBlobs(f func(ch chan<- *blob) error) error {
+func (c *Client) downloadBlobs(ctx context.Context, f func(ch chan<- *blob) error) error {
 	ch := make(chan *blob, 10)
+	done := make(chan struct{})
 	var g errgroup.Group
-	g.Go(func() error { return f(ch) })
+	go func() error {
+		defer close(done)
+		g.Go(func() error { return f(ch) })
 
-	digests := []*pb.Digest{}
-	filenames := map[string]string{}  // map of hash -> output filename
-	modes := map[string]os.FileMode{} // map of hash -> file mode
-	var totalSize int64
-	for b := range ch {
-		filenames[b.Digest.Hash] = b.File
-		modes[b.Digest.Hash] = b.Mode
-		if b.Digest.SizeBytes > c.maxBlobBatchSize || !c.canBatchBlobReads {
-			// This blob individually exceeds the size, have to use this
-			// ByteStream malarkey instead.
-			if err := c.retrieveByteStream(b); err != nil {
-				return err
+		digests := []*pb.Digest{}
+		filenames := map[string]string{}  // map of hash -> output filename
+		modes := map[string]os.FileMode{} // map of hash -> file mode
+		var totalSize int64
```

Comment:
I don't see where this amount ever gets increased during normal blob processing

Reply:
Good point. Perhaps https://github.com/thought-machine/please/pull/770/files#diff-cf387a5ec5198b6baaf902dd7ef5860dR305 ought to be `totalSize += b.Digest.SizeBytes`?